staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go

```diff
@@ -133,23 +117,55 @@ type ExposeServiceOptions struct {
 
 	Recorder genericclioptions.Recorder
 	genericiooptions.IOStreams
+	ExposeInputOptions ExposeInputMetadata
 }
 
-// NewExposeServiceOptions creates a new ExposeServiceOptions and return a pointer to the
-// struct
-func NewExposeServiceOptions(ioStreams genericiooptions.IOStreams) *ExposeServiceOptions {
-	return &ExposeServiceOptions{
-		RecordFlags: genericclioptions.NewRecordFlags(),
-		PrintFlags:  genericclioptions.NewPrintFlags("exposed").WithTypeSetter(scheme.Scheme),
+// exposeServiceFlags is a struct that contains the user input flags to the command.
+type exposeServiceFlags struct {
+	printFlags  *genericclioptions.PrintFlags
+	recordFlags *genericclioptions.RecordFlags
+
+	fieldManager string
+	recorder     genericclioptions.Recorder
+	genericiooptions.IOStreams
+	exposeInputFlags ExposeInputMetadata
+}
+
+// ExposeInputMetadata are the fields which are directly translated from flags to
+// options.
+type ExposeInputMetadata struct {
```

Comment:
If we have this object and this will also be included in Flags and Options, that means we don't fully separate flags from options. I think, it would be better remove this object and adding necessary fields directly into flags and options